### PR TITLE
Add initial tests for exports and run them in mypy and pyright

### PR DESCRIPTION
**Pull request by pakrym-stripe**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,10 +23,23 @@ exclude = '''
...
+  "tests/test_exports.py",
+]
 exclude = ["build", "**/__pycache__"]
 reportMissingTypeArgument = true
 reportUnnecessaryCast = true
 reportUnnecessaryComparison = true
 reportUnnecessaryContains = true
 reportUnnecessaryIsInstance = true
+reportPrivateImportUsage = true
+reportUnnecessaryTypeIgnoreComment = true
```
Comment (pakrym-stripe) [inline pyproject.toml:38]: We might need to use #type: ignore in import tests for negative cases.

File: stripe/multipart_data_generator.py [hunk 2]
```diff
@@ -18,7 +18,10 @@ def __init__(self, chunk_size: int = 1028):
 
     def add_params(self, params):
         # Flatten parameters first
-        params = dict(stripe.api_requestor._api_encode(params))  # type: ignore
+
+        params = dict(
```
Comment (pakrym-stripe) [inline stripe/multipart_data_generator.py:22]: for some reason formatter decided it wants to split some lines

File: stripe/api_resources/quote.py [hunk 3]
```diff
@@ -1567,7 +1567,7 @@ def pdf(
         ...
 
     @util.class_method_variant("_cls_pdf")
-    def pdf(  # type: ignore
+    def pdf(  # pyright: ignore
```
Comment (pakrym-stripe) [inline stripe/api_resources/quote.py:1570]: use pyright ignore consistently

File: tests/test_exports.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# pyright: strict
```
Comment (richardm-stripe) [inline tests/test_exports.py:1]: Love these tests

**Review by richardm-stripe — APPROVED** [review]
Thank you *so much* for doing this. This gives me so much more peace of mind.
